catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/operation/impl/SourceProcessingDetailsImpl.java

```diff
@@ -13,29 +13,57 @@
  */
 package ddf.catalog.operation.impl;
 
+import static org.apache.commons.lang.Validate.notNull;
+
 import ddf.catalog.operation.SourceProcessingDetails;
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
 
 /**
  * The SourceProcessingDetailsImpl class represents a default implementation of a {@link
  * SourceProcessingDetails} to provide {@link Source} warnings.
  */
 public class SourceProcessingDetailsImpl implements SourceProcessingDetails {
 
-  protected List<String> warnings = null;
+  protected List<String> warnings;
 
-  /** Instantiates a new SourceProcessingDetailsImpl. */
-  public SourceProcessingDetailsImpl() {}
+  /** Instantiates a new SourceProcessingDetailsImpl */
+  public SourceProcessingDetailsImpl() {
+    warnings = new ArrayList<>();
+  }
 
   /**
    * Instantiates a new SourceProcessingDetailsImpl.
    *
    * @param warnings the warnings associated with the {@link Source}
    */
   public SourceProcessingDetailsImpl(List<String> warnings) {
+    notNull(
+        warnings, "the constructor of SourceProcessingDetailsImpl does not accept null warnings");
     this.warnings = warnings;
   }
 
+  @Override
+  public boolean equals(Object o) {
+    if (super.equals(o)) {
+      return true;
+    }
```

Comment:
✏️ This is fine, but it hides the fact that all you're really doing is a reference equality comparison. Unless of course this object's inheritance hierarchy changed. Better to be explicit.

Reply:
So you think I should change line 49 back to `if (this==0)`?

Reply:
Oh I see what you did. You tried to be consistent, per my previous comment, but one of the objects actually needed to reference the parent. Okay this is fine.